src/components/catalog/list.js

```diff
@@ -62,61 +63,70 @@ const ArrowUp = styled.div`
   -webkit-transform: rotate(-135deg);
 `;
 
-class List extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      isOpened: false,
-    };
-    this.toggle = this.toggle.bind(this);
-    this.renderItems = this.renderItems.bind(this);
-  }
-
-  toggle() {
-    this.setState({ isOpened: !this.state.isOpened });
-  }
-
-  renderItems(items) {
-    const { padding, onSelect } = this.props;
-
-    return items.map((item, i) => {
-      if (item.children) {
-        return <List key={i} name={item.name} items={item.children} onSelect={onSelect} padding={padding + 16} />;
+export default class List extends React.Component {
+  static propTypes = {
+    name: PropTypes.string.isRequired,
+    methods: PropTypes.array.isRequired,
+    padding: PropTypes.number.isRequired,
+    selectMethod: PropTypes.func.isRequired,
+    selectedMethod: PropTypes.object,
+  };
+
+  static defaultProps = {
+    padding: 10,
+  };
+
+  state = {
+    selectedMethod: this.props.selectedMethod,
+    isOpened: isMethodsContain(this.props.methods, this.props.selectedMethod),
+  };
+
+  toggle = () => this.setState({ isOpened: !this.state.isOpened });
+  select = (selectedMethod) => {
+    this.setState({ selectedMethod });
+    this.props.selectMethod(selectedMethod);
+  };
+  getStyle = (method) => {
+    const { selectedMethod } = this.state;
+    return selectedMethod && method.id === selectedMethod.id ? { background: '#545454', color: '#e2e2e2' } : {};
+  };
+  renderItems = (methods) => {
+    const { padding, selectMethod, selectedMethod } = this.props;
+
+    return methods.map((method, index) => {
+      if (method.children) {
+        return (
+          <List
+            key={index}
+            name={method.name}
+            methods={method.children}
+            padding={padding + 16}
```

Comment:
better to calculate this value once in the constructor, if padding should not change, or at least in the beginning of renderItems, if it can change

Reply:
done